.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,25 @@
+---
+    name: pre-commit
+
+    on:
+      pull_request:
+
+    jobs:
+      lint-pr-title:
+        name: Validate PR titles
+        runs-on: ubuntu-latest
+        steps:
+        - uses: amannn/action-semantic-pull-request@v5
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      pre-commit:
+        runs-on: ubuntu-latest
+        steps:
+        - uses: actions/checkout@v3
+        - uses: actions/setup-python@v4.6.1
```

Comment:
Maintain consistency in version selection. The two default patterns that most workflows should use:

* Mutable version (major)
* Immutable version (SHA)

If you know a particular mutable version is broken and need to pin to a minor or patch, that's fine temporarily. Or if you really want to test every single dependency change and you have Dependabot automating that, you can set exact versions. But use the same semver constraint for all actions, avoid mixing majors and patches unless there's a comment associated with the patch explaining why that exact version is needed.

Exact versions are _not_ pins. As dictated by the behavior git tags, they are mutable references that can change anytime. So most of the time you'll get the same code given the same version. But at no time is that ever guaranteed.

SHAs guarantee immutability but also make it awkward to know which mutable version reference you originated from and how to upgrade to the next one.

See https://github.com/sethvargo/ratchet for some interesting efforts to solve that.

In this case I think you'd be fine with majors and lightening the burden of dependency review.

Reply:
adjusted to use major everywhere. Having ratchet as a pre-commit seems like it would be handy